restli-client/src/main/java/com/linkedin/restli/client/ActionRequestBuilder.java

```diff
@@ -193,7 +195,10 @@ public ActionRequestBuilder(String baseUriTemplate, TypeSpec<V> elementType, Res
     if (_resourceSpec.getRequestMetadata(_name) == null) // old builder code in use
     {
       requestDataSchema = DynamicRecordMetadata.buildSchema(_name, _actionParams.keySet());
-
+      if (_elementType == null)
+      {
+        _elementType = new TypeSpec<>(_elementClass);
```

Comment:
Why are we lazily loading this class-level field in this method? Wouldn't it make more sense to just do so in the constructor?

Reply:
yeah, we have to since this is required for legacy flow only when there were no resourcespec generated. Also, It should be fine since this is builder class not the actual one.

Reply:
Ok, then why isn't it a local variable? It's not clean to have a class-level field which is only initialized and read in one particular method.

Reply:
because with another constructor we accept typespec directly so that we don't have to extract the schema from the class using reflection which is a costly thing

Reply:
Nevermind